packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -82,17 +82,25 @@ function UncontrolledInnerBlocks( props ) {
 		[ clientId ]
 	);
 
-	const { allowSizingOnChildren = false } =
+	const defaultLayoutBlockSupport =
 		getBlockSupport( name, '__experimentalLayout' ) || {};
 
-	const layout = useMemo(
+	const { allowSizingOnChildren = false } = defaultLayoutBlockSupport;
+
+	const defaultLayout = useSetting( 'layout' ) || {};
```

Comment:
Maybe we should use a stable object reference here. Otherwise, `useMemo` will return new values when the layout setting is "falsy".

Reply:
Done!